youtube_dl/extractor/iqiyi.py

```diff
@@ -302,7 +302,7 @@ def _extract_playlist(self, webpage):
         PAGE_SIZE = 50
 
         links = re.findall(
-            r'<a[^>]+class="site-piclist_pic_link"[^>]+href="(http://www\.iqiyi\.com/.+\.html)"',
+            r'<a[^>]+?href="(http://www\.iqiyi\.com/.+\.html)"[^>]+?class="site-piclist_pic_link"[^>]+?>',
```

Comment:
`?` makes no sense here. Arbitrary attribute position should be supported in case this is dynamic.